app/src/common/shared/com/igalia/wolvic/addons/adapters/AddonsListViewAdapter.kt

```diff
@@ -396,10 +398,9 @@ class AddonsManagerAdapter(
         }
     }
 
-    internal fun setWithCrossFadeAnimation(image: ImageView, bitmap: Bitmap, durationMillis: Int = 1700) {
+    internal fun setWithCrossFadeAnimation(image: ImageView, iconDrawable: Drawable, durationMillis: Int = 1700) {
         with(image) {
-            val bitmapDrawable = BitmapDrawable(context.resources, bitmap)
-            val animation = TransitionDrawable(arrayOf(drawable, bitmapDrawable))
+            val animation = TransitionDrawable(arrayOf(drawable, iconDrawable))
```

Comment:
I fail to see how this whole commit modifies anything. It's basically doing the same than before. 

We're just precomputing the iconDrawable, but apart from that I don't see any change in functionality.

Reply:
The main difference with this commit is checking if null and assigning the default icon if custom icon unavailable. Previously, the code did not handle this in the bind function so the previous icon was retained even if new addon was bound to the AddonView item. This led to the incorrect icon bug.

Reply:
I realized I had added the preparatory code but still hadn't fixed the line [I had referred to(main cause of the issue)](https://github.com/Igalia/wolvic/blob/main/app/src/common/shared/com/igalia/wolvic/addons/adapters/AddonsListViewAdapter.kt#L231)! Sorry! I have fixed it in the latest commit [df85788](https://github.com/Igalia/wolvic/pull/391/commits/df85788ed857c86aa43ae318ee5350454c699918)